artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/connect/AMQPBrokerConnection.java

```diff
@@ -0,0 +1,540 @@
+import org.apache.activemq.artemis.spi.core.protocol.ConnectionEntry;
+import org.apache.activemq.artemis.spi.core.remoting.ClientConnectionLifeCycleListener;
+import org.apache.activemq.artemis.spi.core.remoting.ClientProtocolManager;
+import org.apache.activemq.artemis.spi.core.remoting.Connection;
+import org.apache.activemq.artemis.utils.ConfigurationHelper;
+import org.apache.activemq.artemis.utils.UUIDGenerator;
+import org.apache.qpid.proton.amqp.Symbol;
+import org.apache.qpid.proton.amqp.messaging.Source;
+import org.apache.qpid.proton.amqp.messaging.Target;
+import org.apache.qpid.proton.engine.Receiver;
+import org.apache.qpid.proton.engine.Sender;
+import org.apache.qpid.proton.engine.Session;
+import org.jboss.logging.Logger;
+
+public class AMQPBrokerConnection implements ClientConnectionLifeCycleListener, ActiveMQServerQueuePlugin {
+
+   private static final Logger logger = Logger.getLogger(AMQPBrokerConnection.class);
+
+   private final AMQPBrokerConnectConfiguration amqpConfiguration;
+   private final ProtonProtocolManager protonProtocolManager;
+   private final ActiveMQServer server;
+   private final NettyConnector bridgesConnector;
+   private NettyConnection connection;
+   private Session session;
+   AMQPSessionContext sessionContext;
+   ActiveMQProtonRemotingConnection protonRemotingConnection;
+   private volatile boolean started = false;
+   private final AMQPBrokerConnectionManager bridgeManager;
+   private int retryCounter = 0;
+   private volatile ScheduledFuture reconnectFuture;
+   Map<Queue, Sender> senders = new HashMap<>();
+   Map<Queue, Receiver> receivers = new HashMap<>();
+
+   final Executor connectExecutor;
+   final ScheduledExecutorService scheduledExecutorService;
+
+   /** This is just for logging.
+    *  the actual connection will come from the amqpConnection configuration*/
+   String host;
+
+   /** This is just for logging.
+    *  the actual connection will come from the amqpConnection configuration*/
+   int port;
+
+   public AMQPBrokerConnection(AMQPBrokerConnectionManager bridgeManager, AMQPBrokerConnectConfiguration amqpConfiguration,
+                               ProtonProtocolManager protonProtocolManager,
+                               ActiveMQServer server,
+                               NettyConnector bridgesConnector) {
+      this.bridgeManager = bridgeManager;
+      this.amqpConfiguration = amqpConfiguration;
+      this.protonProtocolManager = protonProtocolManager;
+      this.server = server;
+      this.bridgesConnector = bridgesConnector;
+      connectExecutor = server.getExecutorFactory().getExecutor();
+      scheduledExecutorService = server.getScheduledPool();
+   }
+
+   public void stop() {
+      if (connection != null) {
+         connection.close();
+      }
+      ScheduledFuture scheduledFuture = reconnectFuture;
+      reconnectFuture = null;
+      if (scheduledFuture != null) {
+         scheduledFuture.cancel(true);
+      }
+      started = false;
```

Comment:
I see that `started` is volatile but `stop`/`connect` are meant to be called by different threads? 
I would at least guard with single threaded code checks from being started or stopped twice